src/klein/_plating.py

```diff
@@ -25,17 +34,74 @@ def _should_return_json(request):
     return bool(request.args.get(b"json"))
 
 
-def json_serialize(item):
+@inlineCallbacks
+def resolveDeferredObjects(root):
     """
-    A function similar to L{dumps}.
+    Wait on possibly nested L{Deferred}s that represent a JSON
+    serializable object.
+
+    @param root: A JSON-serializable object that may contain
+        L{Deferred}s, or a Deferred that will resolve to a
+        JSON-serializable object
+
+    @return: A L{Deferred} that fires with a L{Deferred}-free version
+        of C{root}, or that fails with the first exception
+        encountered.
     """
-    def helper(unknown):
-        if isinstance(unknown, PlatedElement):
-            return unknown._asJSON()
+
+    result = [None]
+    setResult = partial(setitem, result, 0)
+    stack = [(root, setResult)]
+
+    while stack:
+        mightBeDeferred, setter = stack.pop()
+        obj = yield mightBeDeferred
```

Comment:
A comment here on what things are yielded as themselves is worth it (e.g., async coroutines etc.)